### PR TITLE
All gather expose params

### PR DESCRIPTION
**Pull request by caixunshiren**
### Ticket
#11855 

### Problem description
Expose num workers and num buffers attributes at op level

### CODE REVIEW

**Review by SeanNijjar — CHANGES_REQUESTED** [review]

File: tests/ttnn/unit_tests/operations/test_all_gather_llama_perf_sweep.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+            f"Input shape {input_shape} incompatible with {num_devices} on dim {dim} because some chips will have no tensor",
+        )
+
+    if input_shape == [8, 8, 256, 384] and dim == 1 and layout == ttnn.TILE_LAYOUT and input_dtype == ttnn.bfloat8_b:
+        return True, "Known failure"
+
+    return False, ""
+
+
+def run_all_gather_sharded(
```
Comment (SeanNijjar) [inline tests/ttnn/unit_tests/operations/test_all_gather_llama_perf_sweep.py:62]: Is it possible to just reuse the existing run all gather sharded functions from `test_all_gather` and add the trace mode to that instead?

File: ttnn/cpp/ttnn/operations/ccl/line_all_gather/line_all_gather.hpp [hunk 2]
```diff
@@ -15,15 +15,19 @@ struct ExecuteLineAllGather {
         const ttnn::Tensor& input_tensor,
         const uint32_t dim,
         const uint32_t num_links = 1,
-        const std::optional<ttnn::MemoryConfig>& memory_config = std::nullopt);
+        const std::optional<ttnn::MemoryConfig>& memory_config = std::nullopt,
+        const std::optional<size_t> num_workers = std::nullopt,
+        const std::optional<size_t> num_buffers_per_channel = 2);
```
Comment (SeanNijjar) [inline ttnn/cpp/ttnn/operations/ccl/line_all_gather/line_all_gather.hpp:20]: std::nullopt

File: ttnn/cpp/ttnn/operations/ccl/line_all_gather/line_all_gather.hpp [hunk 3]
```diff
@@ -15,15 +15,19 @@ struct ExecuteLineAllGather {
...
     static ttnn::Tensor invoke(
         const ttnn::Tensor& input_tensor,
         const uint32_t dim,
         const uint32_t cluster_axis,
         const MeshDevice& mesh_device,
         const uint32_t num_links = 1,
-        const std::optional<ttnn::MemoryConfig>& memory_config = std::nullopt);
+        const std::optional<ttnn::MemoryConfig>& memory_config = std::nullopt,
+        const std::optional<size_t> num_workers = std::nullopt,
+        const std::optional<size_t> num_buffers_per_channel = 2);
```
Comment (SeanNijjar) [inline ttnn/cpp/ttnn/operations/ccl/line_all_gather/line_all_gather.hpp:30]: std::nullopt

File: tests/ttnn/unit_tests/operations/test_all_gather_llama_perf_sweep.py [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+import pytest
+from loguru import logger
+import ttnn
+from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import comp_equal, comp_pcc
+from models.utility_functions import skip_for_grayskull, get_devices_for_t3000
+import itertools
+from ttnn import ShardTensorToMesh
+
+
+def is_unsupported_case(input_shape, dim, mem_config, num_devices, num_links, input_dtype, layout):
```
Comment (SeanNijjar) [inline tests/ttnn/unit_tests/operations/test_all_gather_llama_perf_sweep.py:15]: This should be reused from test_all_gather too

**Review by SeanNijjar — CHANGES_REQUESTED** [review]

**Review by SeanNijjar — APPROVED** [review]
Looks good!

**Comment by caixunshiren** [comment]
t3k frequent: https://github.com/tenstorrent/tt-metal/actions/runs/10841532333
all post commit: https://github.com/tenstorrent/tt-metal/actions/runs/10841539264
t3k nightly: https://github.com/tenstorrent/tt-metal/actions/runs/10841543402

**Review by TT-BrianLiu — APPROVED** [review]

**Pull request merged**